deepchecks/tabular/checks/integrity/outlier_detection.py

```diff
@@ -0,0 +1,157 @@
+    between two samples based on its numeric and categorical features.
+    Parameters
+    ----------
+    columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to check, if none are given checks all columns except ignored ones.
+    ignore_columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to ignore, if none given checks based on columns variable
+    num_nearest_neighbors : int, default: 10
+        Number of nearest neighbors to use for outlier detection.
+    extend_parameter: int, default: 3
+        Extend parameter for LoOP algorithm.
+    n_samples : int , default: 50_000
+        number of samples to use for this check.
+    n_to_show : int , default: 5
+        number of data elements with the highest outlier score to show (out of sample).
+    random_state : int, default: 42
+        random seed for all check internals.
+    """
+
+    def __init__(
+            self,
+            columns: Union[Hashable, List[Hashable], None] = None,
+            ignore_columns: Union[Hashable, List[Hashable], None] = None,
+            num_nearest_neighbors: int = 10,
+            extend_parameter: int = 3,
+            n_samples: int = 50_000,
+            n_to_show: int = 5,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        if not isinstance(extend_parameter, int) or extend_parameter <= 0:
+            raise DeepchecksValueError('extend_parameter must be a positive integer')
+        if not isinstance(num_nearest_neighbors, int) or num_nearest_neighbors <= 0:
+            raise DeepchecksValueError('num_nearest_neighbors must be a positive integer')
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.num_nearest_neighbors = num_nearest_neighbors
+        self.extend_parameter = extend_parameter
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.random_state = random_state
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """Run check."""
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        df = select_from_dataframe(dataset.data, self.columns, self.ignore_columns)
+
+        # Calculate distances matrix and retrieve nearest neighbors based on distance matrix.
+        df_cols_for_gower = df[dataset.cat_features + dataset.numerical_features]
+        is_categorical_arr = df_cols_for_gower.columns.map(lambda x: x in dataset.cat_features)
+        dist_matrix = gower_distance.gower_matrix(np.asarray(df_cols_for_gower),
+                                                  cat_features=np.array(is_categorical_arr, dtype=bool))
+        dist_matrix[np.isnan(dist_matrix)] = np.nanmean(dist_matrix)
```

Comment:
I know it's a remote possibility, but let's catch the case when all are nan (think this can happen if the sample is all nulls) and just ignore that sample?